### PR TITLE
setuptools_wrap: Add support for plat-name on macosx

### PR DESCRIPTION
**Pull request by jcfr**
[ci skip]

### CODE REVIEW

**Comment by jcfr** [comment]
@thewtex  This is work in progress .. based on preliminary test, it works quite well.

Do you know if MacOSX wheel are also expected to work on Anaconda ?

File: docs/generators.rst [hunk 1]
```diff
@@ -162,9 +162,116 @@ optimized for building Linux wheels using scikit-build.
...
+command.
+
+For example::
+
+    python setup.py bdist_wheel --plat-name macosx-10.9-x86_64
+
+
+scikit-build also sets the value of `CMAKE_OSX_DEPLOYMENT_TARGET`_ and
+`CMAKE_OSX_ARCHITECTURES`_ option based on the provided platform name. Based on
+the example above, the options used to configured the associated CMake project
```
Comment (thewtex) [inline docs/generators.rst]: `configured` -> `configure`

File: tests/test_cmaker.py [hunk 2]
```diff
@@ -29,6 +29,20 @@ def test_get_python_library():
...
+    assert has_cmake_cache_arg(cmake_args, "FOO", "42")
+    assert not has_cmake_cache_arg(cmake_args, "foo", "42")
+    assert not has_cmake_cache_arg(cmake_args, "FOO", "43")
+    assert not has_cmake_cache_arg(cmake_args, "BAR")
+    assert not has_cmake_cache_arg(cmake_args, "BA")
+    assert not has_cmake_cache_arg(cmake_args, "BAR", None)
+    assert not has_cmake_cache_arg(cmake_args, "BAR", "42")
+    assert has_cmake_cache_arg(cmake_args, "CLIMB")
+    assert has_cmake_cache_arg(cmake_args, "CLIMB", None)
+    assert has_cmake_cache_arg(cmake_args, "CLIMB", "ON")
```
Comment (thewtex) [inline tests/test_cmaker.py]: Climbing.

**Comment by thewtex** [comment]
Testing of the ITK wheels on Mac OSX have worked with Anaconda.

**Pull request merged**